ctapipe/instrument/guess.py

```diff
@@ -24,6 +24,8 @@
     GuessingKey(1855, 28.0): GuessingResult("LST", "LST", "LSTCam", 1),
     GuessingKey(11328, 5.59): GuessingResult("MST", "SCT", "SCTCam", 1),
     # Non-CTA Telescopes
+    GuessingKey(1039, 16.97): GuessingResult("LST", "MAGIC", "MAGICCam", 1),
+    GuessingKey(1039, 0.0): GuessingResult('LST', 'MAGIC', 'MAGICCamMars', 1),
```

Comment:
Could you explain why a focal length of 0 should return another Camera instance? And my impression is, that `MAGICCamMars` has the pixel coordinates in the `EngineeringCameraFrame` aka MAGIC camera coordinates which is not yet supported as input for camera geometries and thus coordinate transformations will be wrong.

Reply:
Yeah sorry my mistake putting a 0 focal lenght and not the same as the other cam, anyway the right camera detected from the MC is MAGICCam, so I can just eliminate the support to the `MAGICCamMars` if you say it's not needed. I just saw it was implemented too and then I added it.

Reply:
There cannot be duplicate keys. The combination of pixels / focal_length have to be unique, otherwise one will not be reachable.

Reply:
If that should arise, we would need a third field in guessing key so telescopes with the same pixel number / focal length could be differentiated. However, where do you need the guessing?

Reply:
EventSources nowadays shoud directly produce instrument info, so now guessing is necessary.

Reply:
ok so better include the `MAGICCam` only that is the "real" one from what I understand. I was trying to analyze MC that have both LST and MAGIC telescopes, and I got `UNKNOWN` camera for MAGIC. Looking at the code the issue was with `event_source` that guess the telescope.

Reply:
I tried to run the code again with the addd of the line `GuessingKey(1039, 16.97): GuessingResult("LST", "MAGIC", "MAGICCam", 1)` and it was working. No `UNKNOWN` camera anymore

Reply:
Should you remove MAGICCamMARS then? Or is it somehow a different camear?

Reply:
No, it is just in the wrong coordinate frame.

Reply:
so it should be removed anyhow - there should only be one "MagicCam". The guess method just helps define the name of a camera, not how it constructed, which is up to the EventSource implementation to do correctly and in the correct `Frame`.

Reply:
So it’s better to remove MagicCamMars to avoid any confusion?

Reply:
yes